src/irc_aliases.lua

```diff
@@ -402,4 +405,59 @@ aliases['config%s+%S+%s+%S+%s*%S*'] = function (ms, c, t, msg, authed)
     end
 end
 
+aliases['qg%s+%S+%s*%S*'] = function (ms, c, t, msg, _, sender)
+    local _, _, action, nick = msg:find('qg%s+(%S+)%s*(%S*)')
+
+    if action == 'status' and nick == '' then
+        local result = ms.irc_quotegrabs.whitelist_status(sender) and 'opted-in' or 'opted-out'
+        ms.irc.privmsg(c, t, ('%s: you are %s for quotegrabs.'):format(sender, result))
+    elseif action == 'status' and nick ~= '' then
+        local result = ms.irc_quotegrabs.whitelist_status(nick) and 'opted-in' or 'opted-out'
+        ms.irc.privmsg(c, t, ('%s is %s for quotegrabs.'):format(nick, result))
+    elseif action == 'optin' then
+        ms.irc_quotegrabs.whitelist_add(sender)
+        ms.irc.privmsg(c, t, sender .. ': Tada! You have been opted-in for quotegrabs.')
```

Comment:
this only happens when the sender has taken this action themselves. The phrasing should probably remain `you have opted-in for quoting` or similar. Similarly for line 422.